notebooks/tfx_pipelines/pipeline/solutions/pipeline_vertex/model.py

```diff
@@ -0,0 +1,313 @@
+    """Returns hyperparameters for building Keras model."""
+    hp = kerastuner.HyperParameters()
+    # Defines search space.
+    hp.Choice("learning_rate", [1e-2, 1e-3, 1e-4], default=1e-3)
+    hp.Int("n_layers", 1, 2, default=1)
+    with hp.conditional_scope("n_layers", 1):
+        hp.Int("n_units_1", min_value=8, max_value=128, step=8, default=8)
+    with hp.conditional_scope("n_layers", 2):
+        hp.Int("n_units_1", min_value=8, max_value=128, step=8, default=8)
+        hp.Int("n_units_2", min_value=8, max_value=128, step=8, default=8)
+
+    return hp
+
+
+def _build_keras_model(
+    hparams: kerastuner.HyperParameters,
+    tf_transform_output: tft.TFTransformOutput,
+) -> tf.keras.Model:
+    """Creates a Keras WideDeep Classifier model.
+    Args:
+      hparams: Holds HyperParameters for tuning.
+      tf_transform_output: A TFTransformOutput.
+    Returns:
+      A keras Model.
+    """
+    deep_columns = [
+        tf.feature_column.numeric_column(
+            key=features.transformed_name(key), shape=()
+        )
+        for key in features.NUMERIC_FEATURE_KEYS
+    ]
+
+    input_layers = {
+        column.key: tf.keras.layers.Input(
+            name=column.key, shape=(), dtype=tf.float32
+        )
+        for column in deep_columns
+    }
+
+    categorical_columns = [
+        tf.feature_column.categorical_column_with_identity(
+            key=features.transformed_name(key),
+            num_buckets=tf_transform_output.num_buckets_for_transformed_feature(
+                features.transformed_name(key)
+            ),
+            default_value=0,
+        )
+        for key in features.CATEGORICAL_FEATURE_KEYS
+    ]
+
+    wide_columns = [
+        tf.feature_column.indicator_column(categorical_column)
+        for categorical_column in categorical_columns
+    ]
+
+    input_layers.update(
+        {
+            column.categorical_column.key: tf.keras.layers.Input(
+                name=column.categorical_column.key, shape=(), dtype=tf.int32
+            )
+            for column in wide_columns
+        }
+    )
+
+    deep = tf.keras.layers.DenseFeatures(deep_columns)(input_layers)
+    for n in range(int(hparams.get("n_layers"))):
+        deep = tf.keras.layers.Dense(
+            units=hparams.get("n_units_" + str(n + 1))
+        )(deep)
+
+    wide = tf.keras.layers.DenseFeatures(wide_columns)(input_layers)
+
+    output = tf.keras.layers.Dense(features.NUM_CLASSES, activation="softmax")(
+        tf.keras.layers.concatenate([deep, wide])
+    )
+
+    model = tf.keras.Model(input_layers, output)
+    model.compile(
+        loss="sparse_categorical_crossentropy",
+        optimizer=tf.keras.optimizers.Adam(lr=hparams.get("learning_rate")),
+        metrics=[tf.keras.metrics.SparseCategoricalAccuracy()],
+    )
+    model.summary(print_fn=absl.logging.info)
+
+    return model
+
+
+# TFX Tuner will call this function.
+def tuner_fn(fn_args: TrainerFnArgs) -> TunerFnResult:
```

Comment:
You can remove this too as it doesn't get called anymore

Reply:
Done!